examples/ct/ct_server/main.go

```diff
@@ -34,26 +36,49 @@ import (
 	"google.golang.org/grpc/naming"
 )
 
-// Global flags that affect all log instances.
-var (
-	serverHostFlag    = flag.String("host", "localhost", "Address to serve CT log requests on")
-	serverPortFlag    = flag.Int("port", 6962, "Port to serve CT log requests on")
-	rpcBackendFlag    = flag.String("log_rpc_server", "localhost:8090", "Backend specification; comma-separated list or etcd service name (if --etcd_servers specified)")
-	rpcDeadlineFlag   = flag.Duration("rpc_deadline", time.Second*10, "Deadline for backend RPC requests")
-	logConfigFlag     = flag.String("log_config", "", "File holding log config in JSON")
-	maxGetEntriesFlag = flag.Int64("maxGetEntriesAllowed", 0, "Max number of entries we allow in a get-entries request (default 50)")
-	etcdServers       = flag.String("etcd_servers", "", "A comma-separated list of etcd servers")
-)
+type config struct {
+	Host                 string
+	Port                 int
+	LogRPCServer         string
+	RPCDeadline          util.ConfigDuration
+	LogConfig            string
+	MaxGetEntriesAllowed int64
+	EtcdServers          string
+}
+
+var configFlag = flag.String("config", "", "Path to JSON config file to load instead of providing config via flags, if provided should be the only flag")
 
 func main() {
+	c := config{}
+	flag.StringVar(&c.Host, "host", "localhost", "Address to serve CT log requests on")
+	flag.IntVar(&c.Port, "port", 6962, "Port to serve CT log requests on")
+	flag.StringVar(&c.LogRPCServer, "log_rpc_server", "localhost:8090", "Backend specification; comma-separated list or etcd service name (if --etcd_servers specified)")
+	flag.DurationVar(&c.RPCDeadline.Duration, "rpc_deadline", time.Second*10, "Deadline for backend RPC requests")
+	flag.StringVar(&c.LogConfig, "log_config", "", "File holding log config in JSON")
+	flag.Int64Var(&c.MaxGetEntriesAllowed, "maxGetEntriesAllowed", 0, "Max number of entries we allow in a get-entries request (default 50)")
+	flag.StringVar(&c.EtcdServers, "etcd_servers", "", "A comma-separated list of etcd servers")
 	flag.Parse()
 
-	if *maxGetEntriesFlag > 0 {
-		ct.MaxGetEntriesAllowed = *maxGetEntriesFlag
+	if *configFlag != "" {
```

Comment:
Have you considered moving this block to a shared function, possibly under the `configs` package I suggested below?